sw/device/lib/dif/dif_sensor_ctrl.c

```diff
@@ -0,0 +1,152 @@
+                      0);
+
+  return kDifOk;
+}
+
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_sensor_ctrl_set_ast_event_trigger(
+    const dif_sensor_ctrl_t *sensor_ctrl, dif_sensor_ctrl_event_idx_t event_idx,
+    dif_toggle_t enable) {
+  if (sensor_ctrl == NULL || is_ast_event_invalid(event_idx)) {
+    return kDifBadArg;
+  };
+
+  uint32_t reg = mmio_region_read32(sensor_ctrl->base_addr,
+                                    SENSOR_CTRL_ALERT_TRIG_REG_OFFSET);
+  reg = bitfield_bit32_write(reg, event_idx, enable);
+  mmio_region_write32(sensor_ctrl->base_addr, SENSOR_CTRL_ALERT_TRIG_REG_OFFSET,
+                      reg);
+
+  return kDifOk;
+}
+
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_sensor_ctrl_set_alert_fatal(
+    const dif_sensor_ctrl_t *sensor_ctrl, dif_sensor_ctrl_event_idx_t event_idx,
+    dif_toggle_t en_fatal) {
+  if (sensor_ctrl == NULL || is_ast_event_invalid(event_idx)) {
+    return kDifBadArg;
+  };
+
+  if (is_locked(sensor_ctrl)) {
+    return kDifLocked;
+  }
+
+  uint32_t reg = mmio_region_read32(sensor_ctrl->base_addr,
+                                    SENSOR_CTRL_FATAL_ALERT_EN_REG_OFFSET);
+  reg = bitfield_bit32_write(reg, event_idx, en_fatal);
+  mmio_region_write32(sensor_ctrl->base_addr,
+                      SENSOR_CTRL_FATAL_ALERT_EN_REG_OFFSET, reg);
+
+  return kDifOk;
+}
+
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_sensor_ctrl_get_recov_events(
+    const dif_sensor_ctrl_t *sensor_ctrl, dif_sensor_ctrl_events_t *events) {
+  if (sensor_ctrl == NULL || events == NULL) {
+    return kDifBadArg;
+  };
+
+  *events = mmio_region_read32(sensor_ctrl->base_addr,
+                               SENSOR_CTRL_RECOV_ALERT_REG_OFFSET);
+
+  return kDifOk;
+};
+
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_sensor_ctrl_clear_recov_event(
+    const dif_sensor_ctrl_t *sensor_ctrl,
+    dif_sensor_ctrl_event_idx_t event_idx) {
+  if (sensor_ctrl == NULL || is_ast_event_invalid(event_idx)) {
+    return kDifBadArg;
+  };
+
+  uint32_t reg = bitfield_bit32_write(0, event_idx, 1);
+  mmio_region_write32(sensor_ctrl->base_addr,
+                      SENSOR_CTRL_RECOV_ALERT_REG_OFFSET, reg);
+
+  return kDifOk;
+}
+
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_sensor_ctrl_get_fatal_events(
+    const dif_sensor_ctrl_t *sensor_ctrl, dif_sensor_ctrl_events_t *events) {
+  if (sensor_ctrl == NULL || events == NULL) {
+    return kDifBadArg;
+  };
+
+  *events = mmio_region_read32(sensor_ctrl->base_addr,
+                               SENSOR_CTRL_FATAL_ALERT_REG_OFFSET);
+
+  return kDifOk;
+};
+
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_sensor_ctrl_get_ast_init_done_status(
+    const dif_sensor_ctrl_t *sensor_ctrl, dif_toggle_t *done) {
+  if (sensor_ctrl == NULL || done == NULL) {
+    return kDifBadArg;
+  };
+
+  uint32_t reg =
+      mmio_region_read32(sensor_ctrl->base_addr, SENSOR_CTRL_STATUS_REG_OFFSET);
+  *done = bitfield_bit32_read(reg, SENSOR_CTRL_STATUS_AST_INIT_DONE_BIT);
```

Comment:
nit: Consider using `dif_bool_to_toggle()` here, since `done` is a `dif_toggle_t`.

Reply:
added